### PR TITLE
env: always expand PATH vars

### LINKED ISSUE #15442: PATH not getting constructed correctly
### Windows Terminal version

1.18.1421.0

### Windows build number

10.0.22621.0

### Other Software

_No response_

### Steps to reproduce

If I launch the new preview terminal, it appears to not be expanding my PATH correctly. This causes apps I regularly use to not be launchable from my terminal.

### Expected Behavior

Notice how the bolded items are properly expanded by the release version of terminal:

1.16.10262.0:
C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\jdk-11.0.16.101-hotspot\bin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Python310\Scripts\;C:\Python310\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\Program Files\Go\bin;C:\Program Files (x86)\dotnet-core-uninstall\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\WINDOWS\SysWOW64\WindowsPowerShell\v1.0\Modules\TShell\TShell\;C:\Program Files\PowerShell\7\;C:\Program Files\Git\cmd;**C:\Users\sotteson\AppData\Local\Microsoft\WindowsApps;C:\Users\sotteson\.dotnet\tools**;C:\Users\sotteson\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\sotteson\AppData\Roaming\npm;D:\data\steve\OneDrive\bin;C:\Users\sotteson\go\bin;C:\Program Files (x86)\GnuWin32\bin;C:\msys64\mingw64\bin

### Actual Behavior

The bolded items are not properly expanded.

1.18.1421.0:
C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\jdk-11.0.16.101-hotspot\bin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Python310\Scripts\;C:\Python310\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\Program Files\Go\bin;C:\Program Files (x86)\dotnet-core-uninstall\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\WINDOWS\SysWOW64\WindowsPowerShell\v1.0\Modules\TShell\TShell\;C:\Program Files\PowerShell\7\;C:\Program Files\Git\cmd;**%USERPROFILE%\AppData\Local\Microsoft\WindowsApps;%USERPROFILE%\.dotnet\tools**;C:\Users\sotteson\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\sotteson\AppData\Roaming\npm;D:\data\steve\OneDrive\bin;%USERPROFILE%\go\bin;C:\Program Files\Go\bin;C:\Program Files (x86)\GnuWin32\bin;C:\msys64\mingw64\bin;

### PR DESCRIPTION
**Pull request by sotteson1**
Make sure we always expand path env vars, even if they're REG_SZ in the registry.

## Detailed Description of the Pull Request / Additional comments
On some systems path vars are REG_SZ instead of REG_EXPAND_SZ. We need to make sure we always expand them. We looked at the system code, and it also makes to sure to always expand them.

## Validation Steps Performed
Built locally and made sure the problem went away. Also stepped through in the debugger to make sure things were working correctly.

Closes #15442

### CODE REVIEW

File: src/inc/til/env.h [hunk 1]
```diff
@@ -313,6 +313,15 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
 
                             if (valueNameSize)
                             {
+                                bool isPathVar = is_path_var(valueName);
```
Comment (zadjii-msft) [inline src/inc/til/env.h:316]:
```suggestion
                                const bool isPathVar = is_path_var(valueName);
```

apparently static analysis got mad at this one

**Review by zadjii-msft — APPROVED** [review]
This seems sensible to me

**Review by lhecker — APPROVED** [review]

**Comment by DHowett** [comment]
Thanks @sotteson1 for fixing this! Great find!

**Comment by sba923** [comment]
One might wonder why on some systems path vars are `REG_SZ` (containing candidates for expansion!) instead of `REG_EXPAND_SZ` in the first place, and why the Windows code does perform expansion on said `REG_SZ`s...

**Comment by DHowett** [comment]
`PATH` is probably the variable changed by the most installers, and so it gets exposed to all sorts of bugs. If I were to guess, expanding it is a defense mechanism against poorly-written software.

**Pull request merged**